compiler/rustc_ty_utils/src/layout/invariant.rs

```diff
@@ -66,12 +66,12 @@ pub(super) fn layout_sanity_check<'tcx>(cx: &LayoutCx<'tcx>, layout: &TyAndLayou
 
     fn check_layout_abi<'tcx>(cx: &LayoutCx<'tcx>, layout: &TyAndLayout<'tcx>) {
         // Verify the ABI mandated alignment and size.
-        let align = layout.backend_repr.inherent_align(cx).map(|align| align.abi);
+        let align = layout.backend_repr.required_align(cx);
         let size = layout.backend_repr.inherent_size(cx);
         let Some((align, size)) = align.zip(size) else {
             assert_matches!(
                 layout.layout.backend_repr(),
-                BackendRepr::Uninhabited | BackendRepr::Memory { .. },
+                BackendRepr::Uninhabited | BackendRepr::Memory { .. } | BackendRepr::Vector { .. },
```

Comment:
I have no recollection of why these two variants are permitted here. But adding `Vector` seems sus as that one is much more like the "scalar" reprs than the others. So why does this make sense?

Reply:
Seems like this came out of a refactor in https://github.com/rust-lang/rust/commit/3b1e535f36ac4c47dc91d0e3394dca72fb86db0c.

Reply:
@RalfJung Because currently we use repr annotations to specify details relevant to the alignment, and the vast majority of alignments we can pick for these types are forgeries: only a few vectors actually have a given alignment for an architecture.